tests/test_core/test_annotationdb.py

```diff
@@ -0,0 +1,633 @@
+    )
+    assert got == expected
+
+    """end provided along with identifier and bio_type"""
+    expected = (
+        "SELECT * FROM GFF WHERE Type == ? AND Attributes like ? AND End < ?",
+        ["CDS", "%RandomIdentifier%", 5000],
+    )
+    got = db._make_sql_query(
+        seq_name=None,
+        bio_type="CDS",
+        identifier="RandomIdentifier",
+        start=None,
+        end=5000,
+    )
+    assert got == expected
+
+    """start and end provided along with identifier and bio_type"""
+    expected = (
+        "SELECT * FROM GFF WHERE Type == ? AND Attributes like ? AND Start >= ? AND End < ?",
+        ["CDS", "%RandomIdentifier%", 0, 5000],
+    )
+    got = db._make_sql_query(
+        seq_name=None, bio_type="CDS", identifier="RandomIdentifier", start=0, end=5000
+    )
+    assert got == expected
+
+    """all five attributes provided"""
+    expected = (
+        "SELECT * FROM GFF WHERE SeqID == ? AND Type == ? AND Attributes like ? AND Start >= ? AND End < ?",
+        ["1", "CDS", "%RandomIdentifier%", 0, 5000],
+    )
+    got = db._make_sql_query(
+        seq_name="1", bio_type="CDS", identifier="RandomIdentifier", start=0, end=5000
+    )
+    assert got == expected
+
+    """check exception when both bio_type and identifier are missing"""
+    import pytest
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=None, end=None
+        )
+
+    """check exception when both bio_type and identifier are missing"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name="1", bio_type=None, identifier=None, start=0, end=1000
+        )
+
+    """check exception when only seq_name is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name="1", bio_type=None, identifier=None, start=None, end=None
+        )
+
+    """check exception when only start is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=0, end=None
+        )
+
+    """check exception when only end is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=None, end=1000
+        )
+
+
+def test_populate_from_file():
+    """Test that the database is populated with the correct
+    number of columns"""
+
+    db = GffAnnotationDb(gff_path)
+    db.db.execute(""" SELECT * FROM GFF """)
+    got = len(list(db.db.fetchall()))
+    parsed_gff = gff_parser(gff_path)
+    expected = len(list(parsed_gff))
+    assert got == expected
+
+
+def test_db_query():
+
+    """Test that the SQL query returns the correct
+    number of rows for different combinations of bio_type/identifier"""
+
+    seqs = load_unaligned_seqs(fasta_path, moltype="dna")
```

Comment:
you don't need `seq`, or `fasta_path` or the associated file. You only need a length! Remove all references to both and just define the sequence length at the top of the test function.